core/Engine/Tweek.Engine/ITweek.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Tweek.Engine.Core;
+using Tweek.Engine.Core.Context;
+using Tweek.Engine.Core.Utils;
+using Tweek.Engine.DataTypes;
+using Tweek.Engine.Drivers.Rules;
+using Tweek.Engine.Rules.Creation;
+using IdentityHashSet = System.Collections.Generic.HashSet<Tweek.Engine.DataTypes.Identity>;
+
+namespace Tweek.Engine
+{
+    public interface ITweek
+    {
+        TweekValuesResult Calculate(
+            ICollection<ConfigurationPath> pathQuery,
+            IdentityHashSet identities, GetLoadedContextByIdentityType context,
+            ConfigurationPath[] includeFixedPaths = null);
+    }
+
+    public delegate IEnumerable<ConfigurationPath> PathExpander(ConfigurationPath path);
+
+    public class TweekRunner : ITweek
```

Comment:
I don't think this file should be called ITweek, it's either tweek.cs or separate to multiple files